network/src/logging.rs

```diff
@@ -10,37 +10,64 @@
 //! use network::logging::*;
 //!
 //! info!(
-//!   network_log(network_events::CONNECTIVITY_MANAGER_LOOP, &NetworkContext::mock())
-//!     .data(network_events::TYPE, network_events::START)
-//!     .field(network_events::EVENT_ID, &5)
+//!   NetworkContextSchema {
+//!     network_context: &NetworkContext::mock()
+//!   },
+//!   field_name = "field",
+//!   "Value is {} message",
+//!   5
 //! );
 //! ```
 //!
 
+use crate::connectivity_manager::DiscoverySource;
 use libra_config::network_id::NetworkContext;
-use libra_logger::StructuredLogEntry;
+use libra_logger::{Schema, StructuredLogEntry};
+use libra_network_address::NetworkAddress;
+use libra_types::PeerId;
 
 /// A helper function to cut down on a bunch of repeated network struct log code
 pub fn network_log(label: &'static str, network_context: &NetworkContext) -> StructuredLogEntry {
     StructuredLogEntry::new_named("network", label)
         .field(network_events::NETWORK_CONTEXT, network_context)
 }
 
+#[derive(Schema)]
+pub struct NetworkSchema<'a> {
+    #[schema(display)]
+    pub discovery_source: Option<&'a DiscoverySource>,
+    pub error: Option<&'a str>,
+    #[schema(display)]
+    pub network_address: Option<&'a NetworkAddress>,
+    pub network_context: &'a NetworkContext,
+    #[schema(display)]
+    pub remote_peer: Option<&'a PeerId>,
```

Comment:
Nit: These fields don't really need to be `pub`.
Nit: Assuming there's just 1 type of error serialized in networking code you could take a reference to that explicit error type instead of to a `str`.

Reply:
Yeah, this was before I was using the setters.  I'm changing it back to a String.  There is not 1 type of error serialized in the networking code, so I just made it a String.